service/service.go

```diff
@@ -56,88 +51,125 @@ func init() {
 // GetStatus gets the current status of the service
 // Returns status code int and status code text, and any connection errors
 func (s *Service) GetStatus(ctx context.Context, req *pb.UserRequest) (*pb.UserResponse, error) {
-	log.RequestService("GetStatus")
+	logger.RequestService("GetStatus")
 
 	// Lock the state for reading and defer unlocks the state before function exits
 	serviceStateLocker.lock.RLock()
 	defer serviceStateLocker.lock.RUnlock()
 
-	log.Info("Service state:", serviceStateMap[serviceStateLocker.currentServiceState])
+	logger.Info("Service state:", serviceStateMap[serviceStateLocker.currentServiceState])
 	if serviceStateLocker.currentServiceState == unavailable {
 		return &pb.UserResponse{
 			Status:  &pb.UserResponse_Code{Code: uint32(codes.Unavailable)},
 			Message: codes.Unavailable.String(),
 		}, nil
 	}
 
-	//Check if mongo clients are found and connected
-	if err := refreshMongoConnection(mongoClientReader); err != nil {
-		log.Error("Failed to ping and reconnect mongo reader server:", err.Error())
-		return &pb.UserResponse{
-			Status:  &pb.UserResponse_Code{Code: uint32(codes.Unavailable)},
-			Message: codes.Unavailable.String(),
-		}, nil
+	if response := refreshDBConnection(); response != nil {
+		return response, nil
 	}
 
-	if err := refreshMongoConnection(mongoClientWriter); err != nil {
-		log.Error("Failed to ping and reconnect mongo writer server:", err.Error())
-		return &pb.UserResponse{
-			Status:  &pb.UserResponse_Code{Code: uint32(codes.Unavailable)},
-			Message: codes.Unavailable.String(),
-		}, nil
+	return &pb.UserResponse{
+		Status:  &pb.UserResponse_Code{Code: uint32(codes.OK)},
+		Message: codes.OK.String(),
+	}, nil
+}
+
+// CreateUser creates a new user document and inserts it to user DB
+func (s *Service) CreateUser(ctx context.Context, req *pb.UserRequest) (*pb.UserResponse, error) {
+	logger.RequestService("CreateUser")
+
+	// get User Object
+	user := req.GetUser()
+	if user == nil {
+		logger.Error(errNilRequestUser.Error())
+		return nil, status.Error(codes.InvalidArgument, errNilRequestUser.Error())
+	}
+
+	// validate fields in user object
+	if str, err := validateUser(user); err != nil {
+		logger.Error(str, ":", err.Error())
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	// generate uuid
+	id, err := generateUUID()
```

Comment:
How do you know that this will be synchronized. Meaning that there is no chance we will have 2 or more users with the same UUID?